src/emu/diimage.h

```diff
@@ -205,6 +205,7 @@ class device_image_interface : public device_interface
 	const software_info *software_entry() const { return m_software_info_ptr; }
 	const software_part *part_entry() const { return m_software_part_ptr; }
 	const char *software_list_name() const { return m_software_list_name.c_str(); }
+	bool loaded_through_softlist() const { return m_software_info_ptr != nullptr; }
```

Comment:
Will `m_software_info_ptr` always get reset to `nullptr` if softlist media is removed and non-softlist media loaded?  If not, a stale value could cause a problem.